posts/2023-11-05-telegram-bot-for-file-sharing-using-quarkus.adoc

```diff
@@ -0,0 +1,546 @@
+
+Additionally, we have to manually add https://github.com/rubenlagus/TelegramBots - a library for creating telegram bots, and https://github.com/minio/minio-java - MinIO java client.
+
+While in the root directory of our new project, run this command -
+```bash
+quarkus create app com.example:backend \
+--extension=quarkus-mongodb-panache,quarkus-resteasy \
+--gradle
+```
+
+We can already run our application with the default implementation. To do so, navigate to the `/backend` directory and execute the command to start our application.
+```bash
+quarkus dev
+```
+
+Now, you can open the recommended welcome page at http://localhost:8080/
+
+image::welcome_page.png[,600]
+and http://localhost:8080/heloo default controller method.
+
+image::controller_hello.png[,600]
+
+It was quite easy, right?
+
+Before we dive into coding the bot's logic, we need to register our bot and obtain a username and token. To do this, please follow the official instructions provided here: https://core.telegram.org/bots/tutorial.
+
+It's worth noting that there isn't even a main class that we're used to in the source code, which can sometimes be considered boilerplate. However, let's go ahead and create one explicitly now.
+[source,java]
+----
+package com.example;
+
+import io.quarkus.runtime.Quarkus;
+import io.quarkus.runtime.QuarkusApplication;
+import io.quarkus.runtime.annotations.QuarkusMain;
+import org.jboss.logging.Logger;
+import org.telegram.telegrambots.bots.TelegramLongPollingBot;
+import org.telegram.telegrambots.meta.TelegramBotsApi;
+import org.telegram.telegrambots.meta.exceptions.TelegramApiException;
+import org.telegram.telegrambots.updatesreceivers.DefaultBotSession;
+
+@QuarkusMain
+public class Main {
+
+    public static void main(String... args) {
+        Quarkus.run(MyApp.class, args);
+    }
+
+    public static class MyApp implements QuarkusApplication {
+        private static final Logger LOGGER = Logger.getLogger(MyApp.class);
+        private final TelegramLongPollingBot telegramLongPollingBot;
+
+        public MyApp(TelegramLongPollingBot telegramLongPollingBot) {
+            this.telegramLongPollingBot = telegramLongPollingBot;
+        }
+
+        @Override
+        public int run(String... args) {
+            try {
+                TelegramBotsApi telegramBotsApi = new TelegramBotsApi(DefaultBotSession.class);
+                telegramBotsApi.registerBot(telegramLongPollingBot);
+            } catch (TelegramApiException e) {
+                LOGGER.errorf(e, e.getMessage());
+            }
+            Quarkus.waitForExit();
+            return 0;
+        }
+    }
+}
+----
+
+Dependency injection in Quarkus is a CDI-based dependency injection - https://quarkus.io/guides/cdi. In the case of only one defined constructor, Quarkus uses constructor injection. To satisfy the constructor parameter let's create a corresponding bean using the method return type by [.yellow]#@Produce# annotation.
```

Comment:
I think a bit of context of which dependency is going to be injected is missing here, at the beginning of the paragraph